### DISCUSSION TITLE
Setting Media Library as the default upload manager

### DISCUSSION THREAD

**Discussion by scvn1** [awcodes/filament-tiptap-editor#117] [Q&A]
All images are being handled with the default Filament uploader instead of the Spatie Media Library integration that I've installed into Filament. Works fine with any other component but not with TipTap.

How would I set the default file upload handler in the TipTap plugin to the Spatie Media Library so I see all files being tracked in the media table?

**Comment by awcodes** [comment]
You would need to make your own `MediaAction` then set it in the config.

**Comment by awcodes** [comment]
See https://github.com/awcodes/filament-tiptap-editor/blob/2.x/src/Actions/MediaAction.php for an example.

**Reply by lpeterke** [reply]
Hi there! I am trying to implement this and it doesn't seem to work. I simply switched out the FileUpload in the Form with the SpatieMediaLibraryFileUpload and removed the uneccessary saveUploadedFileUsing() callback, as this is defined the SpatieMediaLibraryFileUpload class itself.

Everything works, however I get an exception in the action() callback of the MediaAction, because the 'src' attribute is not set. And I can't seem to find a way to reach for the url of my newly uploaded File to pass it back to the tiptap editor.

**Reply by awcodes** [reply]
You will need to make a custom media action if you want to change any of the fields. And you definitely need the saveUploadedFileUsing to be custom. It will not work with the default saving of FileUpload or SpatieFileUpload.

**Reply by lpeterke** [reply]
Thanks for the reply! I will try to work my way through this. 

The whole thing seems to be a bit tricky, as the Spatie Media Library always requires a model to associate the images with. And when creating a NEW blog post for example, any uploaded images via TipTap of course don't have a model target, as the post has not been created yet... :D So yeah, probably need something more advanced to get this working...

**Reply by awcodes** [reply]
And that's why I don't use spatie's media library. :) Having to have an associated model has always been a no go for me when working with media.
